src/main/java/com/dimensiondelvers/dimensiondelvers/block/BlockFamilyHelper.java

```diff
@@ -0,0 +1,165 @@
+        return slab;
+    }
+
+    public Supplier<Block> getStairs() {
+        return stairs;
+    }
+
+    public Supplier<Block> getButton() {
+        return button;
+    }
+
+    public Supplier<Block> getPressurePlate() {
+        return pressurePlate;
+    }
+
+    public Supplier<Block> getFence() {
+        return fence;
+    }
+
+    public Supplier<Block> getWall() {
+        return wall;
+    }
+
+    public Supplier<Block> getTrapdoor() {
+        return trapdoor;
+    }
+
+    public BlockFamily getFamily() {
+        if (blockFamily == null) {
+            blockFamily = new BlockFamily.Builder(block.get())
+                    .slab(slab.get())
+                    .stairs(stairs.get())
+                    .button(button.get())
+                    .pressurePlate(pressurePlate.get())
+                    .wall(wall.get())
+                    .fence(fence.get())
+                    .trapdoor(trapdoor.get())
+                    .getFamily();
+        }
+        return blockFamily;
+    }
+
+    public ResourceLocation getBaseResourceLocation() {
+        return ResourceLocation.fromNamespaceAndPath(MOD_ID, blockId);
+    }
+
+    public static class Builder {
+
+        private String blockId;
+        private Supplier<Block> block;
+        private Supplier<Block> slab;
+        private Supplier<Block> stairs;
+        private Supplier<Block> button;
+        private Supplier<Block> pressurePlate;
+        private Supplier<Block> wall;
+        private Supplier<Block> fence;
+        private Supplier<Block> trapdoor;
+
+        public Builder withBlockId(String blockId) {
+            this.blockId = blockId;
+            return this;
+        }
+
+        public Builder withBlock(Supplier<Block> block) {
+            this.block = block;
+            return this;
+        }
+
+        public Builder withSlab(Supplier<Block> slab) {
+            this.slab = slab;
+            return this;
+        }
+
+        public Builder withStairs(Supplier<Block> stairs) {
+            this.stairs = stairs;
+            return this;
+        }
+
+        public Builder withButton(Supplier<Block> button) {
+            this.button = button;
+            return this;
+        }
+
+        public Builder withPressurePlate(Supplier<Block> pressurePlate) {
+            this.pressurePlate = pressurePlate;
+            return this;
+        }
+
+        public Builder withFence(Supplier<Block> fence) {
+            this.fence = fence;
+            return this;
+        }
+
+        public Builder withWall(Supplier<Block> wall) {
+            this.wall = wall;
+            return this;
+        }
+
+        public Builder withTrapdoor(Supplier<Block> trapdoor) {
+            this.trapdoor = trapdoor;
+            return this;
+        }
+
+        public BlockFamilyHelper createBuildBlockHelper() {
+            return new BlockFamilyHelper(blockId, block, slab, stairs, button, pressurePlate, wall, fence, trapdoor);
```

Comment:
It might be worth adding a check here that all the values have been populated, just to fail early in case one was missed. Although it will be clear eventually when a NPE is raised.